apps/charterafrica/src/lib/data/common/index.js

```diff
@@ -349,8 +342,9 @@ export async function getPageProps(context, api) {
       })
     )) || [];
   const processPage = processPageFunctionsMap[page.slug];
+  let processedPage = {};
   if (processPage) {
-    await processPage(page, api);
+    processedPage = await processPage(page, api);
```

Comment:
`processPage` can & should modify the page or why do we need to create a copy?

Reply:
The other workaround if we need to mutate page object directly is do something like

```js
  blocks[0] = featuredPost;
  blocks[1] = news;
```

because we need to process a block which has a slug of `featured-post` but data structure is not in a way we want.